### PR TITLE
[module][Kotlin] Fix `findView` on fabric

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Fixes `findView` function on fabric. 

# Test Plan

- camera module is now working on fabric

### CODE REVIEW

File: packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt [hunk 1]
```diff
@@ -264,8 +265,13 @@ class AppContext(
   @Suppress("UNCHECKED_CAST")
   @UiThread
   fun <T : View> findView(viewTag: Int): T? {
-    val rnUIManager = reactContextHolder.get()?.getNativeModule(UIManagerModule::class.java)
-    return rnUIManager?.resolveView(viewTag) as? T
+    return if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
```
Comment (tsapeta) [inline packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt:268]: Let's use `RN_FABRIC_ENABLED` as it works on both platforms 😉
```suggestion
    return if (BuildConfig.RN_FABRIC_ENABLED) {
```

**Review by tsapeta — APPROVED** [review]

File: packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt [hunk 2]
```diff
@@ -264,8 +265,13 @@ class AppContext(
...
+      val fabricUIManager = reactContextHolder.get()?.getJSIModule(JSIModuleType.UIManager) as? com.facebook.react.bridge.UIManager
+      fabricUIManager?.resolveView(viewTag) as? T
+    } else {
+      val paperUIManager = reactContextHolder.get()?.getNativeModule(com.facebook.react.uimanager.UIManagerModule::class.java)
+      paperUIManager?.resolveView(viewTag) as? T
+    }
```
Comment (Kudo) [inline packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt:274]:
```suggestion
    val context = reactContextHolder.get() ?: return null
    return UIManagerHelper.getUIManagerForReactTag(context, viewTag)?.resolveView(viewTag) as? T
```

**Review by Kudo — CHANGES_REQUESTED** [review]
i would prefer to use `UIManagerHelper` for this.

File: packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt (hunk 1 above)
Comment (lukmccall) [reply packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt:268]: This flag isn't available in the `BuildConfig` 🤷

File: packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt [hunk 3]
```diff
@@ -7,9 +7,11 @@ import android.view.View
 import androidx.annotation.MainThread
 import androidx.annotation.UiThread
 import androidx.appcompat.app.AppCompatActivity
+import com.facebook.react.bridge.JSIModuleType
```
Comment (Kudo) [inline packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt:10]:
```suggestion
```

File: packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt [hunk 4]
```diff
@@ -7,9 +7,11 @@ import android.view.View
...
 import com.facebook.react.bridge.ReactApplicationContext
 import com.facebook.react.turbomodule.core.CallInvokerHolderImpl
-import com.facebook.react.uimanager.UIManagerModule
+import com.facebook.react.uimanager.UIManagerHelper
+import expo.modules.BuildConfig
```
Comment (Kudo) [inline packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt:14]:
```suggestion
```

**Review by Kudo — APPROVED** [review]
nit: remove unused imports

**Pull request merged**